File: python/svix/models/event_type_in.py
# this file is @generated
import typing as t

from .common import BaseModel


class EventTypeIn(BaseModel):
    name: str
    """The event type's name"""

    description: str

    archived: t.Optional[bool] = None

    deprecated: t.Optional[bool] = None

    schemas: t.Optional[t.Dict[str, t.Any]] = None
    """The schema for the event type for a specific version as a JSON schema."""

    group_name: t.Optional[str] = None
    """The event type group's name"""

    feature_flags: t.Optional[t.List[str]] = None

File: csharp/Svix/Models/EventTypeIn.cs
// this file is @generated
using Newtonsoft.Json;

namespace Svix.Models
{
    public class EventTypeIn : BaseModel
    {
        [JsonProperty("archived")]
        public bool? Archived { get; set; } = null;

        [JsonProperty("deprecated")]
        public bool? Deprecated { get; set; } = null;

        [JsonProperty("description", Required = Required.Always)]
        public required string Description { get; set; }

        [JsonProperty("featureFlag")]
        public string? FeatureFlag { get; set; } = null;

        [JsonProperty("groupName")]
        public string? GroupName { get; set; } = null;

        [JsonProperty("name", Required = Required.Always)]
        public required string Name { get; set; }

        [JsonProperty("schemas")]
        public Object? Schemas { get; set; } = null;
    }
}
